mods/ra/maps/siberian-conflict-1-fresh-tracks/fresh-tracks.lua

```diff
@@ -0,0 +1,275 @@
+	{ SouthEntry.Location, SouthPath1.Location, Patrol2.Location, SouthPath2.Location, SoutheastBridge.Location, SouthPath3.Location, ExitEast.Location }
+}
+
+BaseConvoyPath = { BasePath1, CenterPath2, Patrol1, Patrol7, Patrol6, NortheastBridge, NorthwestPath3, ExitNorth }
+FinalConvoyPath = { Patrol7, Patrol6, Patrol5, FinalConvoyPath, ExitEast }
+
+OpeningMoves = function()
+	Utils.Do(Squad1, function(actor)
+		actor.AttackMove(AlliedBase.Location)
+	end)
+
+	Utils.Do(Squad2, function(actor)
+		actor.AttackMove(AlliedBase.Location)
+	end)
+
+	PatrolMammoth.Patrol(PatrolMammothPath, true, 10)
+end
+
+AttackWaveDelays =
+{
+	easy = { DateTime.Seconds(40), DateTime.Seconds(80) },
+	normal = { DateTime.Seconds(30), DateTime.Seconds(60) },
+	hard = { DateTime.Seconds(20), DateTime.Seconds(40) }
+}
+
+AttackWaves = function()
+	local attackpath = Utils.Random(AttackPaths)	
+	local attackers = Reinforcements.Reinforce(USSR, Utils.Random(AttackWaveUnits), { attackpath[1] })
+	Utils.Do(attackers, function(unit)
+		Trigger.OnAddedToWorld(unit, function()
+			unit.AttackMove(AlliedBase.Location)
+			unit.Hunt()
+		end)
+	end)
+
+	Trigger.AfterDelay(Utils.RandomInteger(AttackWaveDelay[1], AttackWaveDelay[2]), AttackWaves)
+end
+
+ConvoyWaves = 
+{
+	easy = 2,
+	normal = 3,
+	hard = 4
+}
+ConvoysPassed = 0
+
+SendConvoys = function()
+	ConvoysSent = true
+	Media.PlaySpeechNotification(Allies, "ConvoyApproaching")
+	local path = Utils.Random(ConvoyPaths)
+	local units = Reinforcements.Reinforce(USSR, Utils.Random(ConvoyUnits), { path[1] })
+	local lastWaypoint = path[#path]
+	Utils.Do(units, function(unit)
+		Trigger.OnAddedToWorld(unit, function()
+			if unit.Type == "truk" then
+				Utils.Do(path, function(waypoint)
+					unit.Move(waypoint)
+				end)
+
+				Trigger.OnIdle(unit, function()
+					unit.Move(lastWaypoint)
+				end)
+			else
+				unit.Patrol(path)
+				Trigger.OnIdle(unit, function()
+					unit.AttackMove(lastWaypoint)
+				end)
+			end
+		end)
+	end)
+
+	ConvoysPassed = ConvoysPassed + 1
+	if ConvoysPassed <= ConvoyWaves[Map.LobbyOption("difficulty")] then
+		Trigger.AfterDelay(DateTime.Seconds(90), SendConvoys)
+	else
+		FinalTrucks()
+	end
+end
+
+FinalTrucks = function()
+	Trigger.AfterDelay(DateTime.Minutes(1), function()
+		Media.PlaySpeechNotification(Allies, "ConvoyApproaching")
+		local baseConvoy = Reinforcements.Reinforce(USSR, BaseConvoyUnits, { BaseEntry.Location })
+		Utils.Do(BaseConvoyPath, function(waypoint)
+			Utils.Do(baseConvoy, function(actor)
+				Trigger.OnAddedToWorld(actor, function(a)
+					a.Move(waypoint.Location)
+				end)
+			end)
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Minutes(2), function()
+		Media.PlaySpeechNotification(Allies, "ConvoyApproaching")
+		local finalConvoy = Reinforcements.Reinforce(USSR, FinalConvoyUnits, { NorthEntry.Location })
+		Utils.Do(FinalConvoyPath, function(waypoint)
```

Comment:
Why isn't this the path given to `Reinforcements.Reinforce`? Because units should not attack-move? Either way, we're currently adding on `OnAddedToWorld` trigger for every waypoint.
Better would be (if we don't give the path to `Reinforcements.Reinforce`):
```
Utils.Do(finalConvoy, function(actor)
	Trigger.OnAddedToWorld(actor, function(a)
		Utils.Do(FinalConvoyPath, function(waypoint)
			a.Move(waypoint.Location)
		end)
	end)
end)
```